jobs/upload-brew.Jenkinsfile

```
@@ -0,0 +1,136 @@
+            image: cosa_img,
+            serviceAccount: "jenkins",
+            secrets: ["brew-keytab", "brew-ca:ca.crt:/etc/pki/ca.crt",
+                      "koji-conf:koji.conf:/etc/koji.conf",
+                      "krb5-conf:krb5.conf:/etc/krb5.conf"]) {
+    timeout(time: 240, unit: 'MINUTES') {
+    try {
+        stage('Fetch Metadata') {
+            def ref = pipeutils.get_source_config_ref_for_stream(pipecfg, params.STREAM)
+            def variant = stream_info.variant ? "--variant ${stream_info.variant}" : ""
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa init --branch ${ref} ${variant} ${pipecfg.source_config.url}
+                cosa buildfetch --build=${params.VERSION} \
+                    --arch=all --url=s3://${s3_stream_dir}/builds \
+                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                    --file "coreos-assembler-config-git.json"
+            """)
+        }
+        def builtarches = shwrapCapture("""
+                          cosa shell -- cat builds/builds.json | \
+                              jq -r '.builds | map(select(.id == \"${params.VERSION}\"))[].arches[]'
+                          """).split() as Set
+        assert builtarches.contains("x86_64"): "The x86_64 architecture was not in builtarches."
+        if (!builtarches.containsAll(basearches)) {
+            if (params.ALLOW_MISSING_ARCHES) {
+                warn("Some requested architectures did not successfully build!")
+                basearches = builtarches.intersect(basearches)
+            } else {
+                echo "ERROR: Some requested architectures did not successfully build"
+                echo "ERROR: Detected built architectures: $builtarches"
+                echo "ERROR: Requested base architectures: $basearches"
+                currentBuild.result = 'FAILURE'
+                return
+            }
+        }
+
+        stage('Brew Upload') {
+            def tag = pipecfg.streams[params.STREAM].brew_tag
+            for (arch in basearches) {
+                shwrap("""
+                    coreos-assembler koji-upload \
+                        upload --reserve-id \
+                        --keytab "/run/kubernetes/secrets/brew-keytab/brew.keytab" \
+                        --build ${params.VERSION} \
+                        --retry-attempts 6 \
+                        --buildroot builds \
+                        --owner ${brew_principal} \
+                        --profile ${brew_profile} \
+                        --tag ${tag} \
+                        --arch "$arch"
+                """)
+            }
+        }
+        currentBuild.result = 'SUCCESS'
+
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        throw e
+    } finally {
+        def message = "[${params.STREAM}][${params.VERSION}][${params.ARCH}]"
```

```suggestion
        def message = "[${params.STREAM}][${basearches.join(' ')}] (${params.VERSION})"
```

Let's make this more like the release job.